sim/common/vanilla/item_effects.go

```diff
@@ -626,5 +627,45 @@ func init() {
 		}
 	})
 
+	core.NewItemEffect(MarkOfTheChosen, func(agent core.Agent) {
+		character := agent.GetCharacter()
+		statIncrease := float64(25)
+		markProcChance := 0.02
+
+		procAura := character.RegisterAura(core.Aura{
+			Label:    "Mark of the Chosen Effect",
+			ActionID: core.ActionID{SpellID: 21970},
+			Duration: time.Minute,
+			OnGain: func(aura *core.Aura, sim *core.Simulation) {
+				character.AddStatsDynamic(sim, stats.Stats{
+					stats.Stamina:   statIncrease,
+					stats.Agility:   statIncrease,
+					stats.Strength:  statIncrease,
+					stats.Intellect: statIncrease,
+					stats.Spirit:    statIncrease,
+				})
+			},
+			OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+				character.AddStatsDynamic(sim, stats.Stats{
+					stats.Stamina:   -statIncrease,
+					stats.Agility:   -statIncrease,
+					stats.Strength:  -statIncrease,
+					stats.Intellect: -statIncrease,
+					stats.Spirit:    -statIncrease,
+				})
+			},
+		})
+
+		core.MakePermanent(character.RegisterAura(core.Aura{
+			Label:    "Mark of the Chosen",
+			ActionID: core.ActionID{SpellID: 21969},
+			OnSpellHitTaken: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if result.Landed() && spell.ProcMask.Matches(core.ProcMaskMelee) && sim.RandomFloat("Mark of the Chosen") < markProcChance {
```

Comment:
🔨 Something I didn't realize until recently is you can use `sim.Proc(markProcChance, "Mark of the Chosen")` and it will do the randomfloat comparison for you. Just a bit nicer and something I want to clean up in code I've touched